plugins/components/collider/Collider.hpp

```diff
@@ -0,0 +1,143 @@
+        }
+
+        /**
+         * @brief Serialize the collider data
+         * 
+         * Serializes the width and height into a byte vector for transmission or storage.
+         * 
+         * @return std::vector<uint8_t> Serialized data.
+         */
+        std::vector<uint8_t> serialize() override {
+            std::vector<uint8_t> data;
+            appendToData(data, width);
+            appendToData(data, height);
+            return data;
+        };
+
+        /**
+         * @brief Deserialize the collider data
+         * 
+         * Deserializes the width and height from the provided byte vector.
+         * 
+         * @param data The byte vector containing serialized data.
+         * @throws std::runtime_error If the data size is invalid.
+         */
+        void deserialize(std::vector<uint8_t> &data) override {
+            if (data.size() != getConstantSize())
+                throw std::runtime_error("Invalid data size for Collider component");
+
+            width = *reinterpret_cast<float *>(&data[0]);
+            height = *reinterpret_cast<float *>(&data[sizeof(float)]);
+        };
+
+        /**
+         * @brief Get the fixed size of the serialized data
+         * 
+         * @return size_t Size in bytes
+         */
+        size_t getSize() const override {
+            return getConstantSize();
+        }
+
+        /**
+         * @brief Adds the Collider component to an entity.
+         * 
+         * @param to The entity to add the component to.
+         * @param engine The game engine.
+         * @param args The arguments to pass to the component constructor.
+         * 
+         * @note The arguments should be a tuple containing the width and height.
+         */
+        void addTo(ECS::Entity &to, Engine::GameEngine &engine, std::vector<std::any> args) override {
+            if (args.size() != 2)
+                throw std::runtime_error("Invalid number of arguments for Collider component");
+
+            float width = std::any_cast<float>(args[0]);
+            float height = std::any_cast<float>(args[1]);
+
+            auto collider = engine.newComponent<Components::Collider>(width, height);
+            engine.getRegistry().componentManager().addComponent<Components::Collider>(to, std::move(collider));
+        };
+
+        /**
+         * @brief Adds the Collider component to an entity from a configuration setting.
+         * 
+         * @param to The entity to add the component to.
+         * @param engine The game engine.
+         * @param config The configuration setting to extract the component data from.
+         */
+        void addTo(ECS::Entity &to, Engine::GameEngine &engine, libconfig::Setting &config) override {
+            float width = 0.0f, height = 0.0f;
+
+            if (!config.lookupValue("width", width) || !config.lookupValue("height", height)) {
+                throw std::invalid_argument("Failed to retrieve values for 'width' and 'height'");
```

Comment:
Maybe instead of throwing, we could just print an error message and defaulting the variables ?